src/axom/core/ItemCollection.hpp

```diff
@@ -318,6 +325,7 @@ class ItemCollection<T>::const_iterator_adaptor
 
 private:
   const CollectionType* m_collection {nullptr};
+  std::size_t counter {0};
```

Comment:
Needed to add a counter private member here to be able to get the python iteration working. Not certain if this was the cleanest way to get things working. 

Nanobind does have an API function `make_iterator` to add python iterable support to a class, but that requires the class to have a public constructor & destructor (not compatible with Sidre classes).